File: app/src/main/java/nz/co/codebros/quakesnz/GeonetService.java
package nz.co.codebros.quakesnz;

import io.reactivex.Single;
import nz.co.codebros.quakesnz.model.FeatureCollection;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by leandro on 9/07/15.
 */
public interface GeonetService {

    @GET("quake/{publicID}")
    Single<FeatureCollection> getQuake(@Path("publicID") String publicID);

    @GET("quake")
    Single<FeatureCollection> getQuakes(@Query("MMI") int mmi);
}

File: core/src/main/java/nz/co/codebros/quakesnz/core/service/GeonetService.kt
package nz.co.codebros.quakesnz.core.service

import io.reactivex.Single
import nz.co.codebros.quakesnz.core.data.FeatureCollection
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GeonetService {

    @GET("quake/{publicID}")
    fun getQuake(@Path("publicID") publicID: String): Single<FeatureCollection>

    @GET("quake")
    fun getQuakes(@Query("MMI") mmi: Int): Single<FeatureCollection>
}
